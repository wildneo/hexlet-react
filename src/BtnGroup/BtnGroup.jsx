import React from 'react';
import cn from 'classnames';

export default class BtnGroup extends React.Component {
  constructor(props) {
    super(props);
    this.state = { active: null };

    this.leftHandler = () => this.setActive('left');
    this.rightHandler = () => this.setActive('right');
  }

  setActive(active) {
    this.setState({ active });
  }

  render() {
    const { active } = this.state;
    const sharedClasses = ['btn', 'btn-secondary'];
    const leftStyle = cn(...sharedClasses, 'left', { active: active === 'left' });
    const rightStyle = cn(...sharedClasses, 'right', { active: active === 'right' });

    return (
      <div className="btn-group" role="group">
        <button onClick={this.leftHandler} type="button" className={leftStyle}>Left</button>
        <button onClick={this.rightHandler} type="button" className={rightStyle}>Right</button>
      </div>
    );
  }
}
