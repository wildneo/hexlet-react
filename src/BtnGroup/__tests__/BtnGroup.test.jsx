import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import BtnGroup from '../BtnGroup';

Enzyme.configure({ adapter: new Adapter() });

test('BtnGroup 1', () => {
  const wrapper = mount(<BtnGroup />);
  expect(wrapper.render()).toMatchSnapshot();

  wrapper.find('.left').simulate('click');
  expect(wrapper.render()).toMatchSnapshot();

  wrapper.find('.right').simulate('click');
  expect(wrapper.render()).toMatchSnapshot();

  wrapper.find('.left').simulate('click');
  expect(wrapper.render()).toMatchSnapshot();

  wrapper.find('.left').simulate('click');
  expect(wrapper.render()).toMatchSnapshot();
});
