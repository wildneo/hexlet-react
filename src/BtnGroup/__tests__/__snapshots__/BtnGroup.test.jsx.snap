// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BtnGroup 1 1`] = `
<div
  class="btn-group"
  role="group"
>
  <button
    class="btn btn-secondary left"
    type="button"
  >
    Left
  </button>
  <button
    class="btn btn-secondary right"
    type="button"
  >
    Right
  </button>
</div>
`;

exports[`BtnGroup 1 2`] = `
<div
  class="btn-group"
  role="group"
>
  <button
    class="btn btn-secondary left active"
    type="button"
  >
    Left
  </button>
  <button
    class="btn btn-secondary right"
    type="button"
  >
    Right
  </button>
</div>
`;

exports[`BtnGroup 1 3`] = `
<div
  class="btn-group"
  role="group"
>
  <button
    class="btn btn-secondary left"
    type="button"
  >
    Left
  </button>
  <button
    class="btn btn-secondary right active"
    type="button"
  >
    Right
  </button>
</div>
`;

exports[`BtnGroup 1 4`] = `
<div
  class="btn-group"
  role="group"
>
  <button
    class="btn btn-secondary left active"
    type="button"
  >
    Left
  </button>
  <button
    class="btn btn-secondary right"
    type="button"
  >
    Right
  </button>
</div>
`;

exports[`BtnGroup 1 5`] = `
<div
  class="btn-group"
  role="group"
>
  <button
    class="btn btn-secondary left active"
    type="button"
  >
    Left
  </button>
  <button
    class="btn btn-secondary right"
    type="button"
  >
    Right
  </button>
</div>
`;
